src/utils/mutateInternal.ts

```diff
@@ -0,0 +1,239 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  Entry,
+  MutateCallback,
+  MutateOptions,
+  PartialFailureError,
+} from '../table';
+import {OperationMetricsCollector} from '../client-side-metrics/operation-metrics-collector';
+import {
+  DEFAULT_BACKOFF_SETTINGS,
+  getNextDelay,
+  populateAttemptHeader,
+  RETRYABLE_STATUS_CODES,
+  TabularApiSurface,
+} from '../tabular-api-surface';
+import {ServiceError} from 'google-gax';
+import {google} from '../../protos/protos';
+import * as is from 'is';
+import {Mutation} from '../mutation';
+import arrify = require('arrify');
+
+export function mutateInternal(
```

Comment:
this should probably have [a comment like getRowsInternal](https://github.com/googleapis/nodejs-bigtable/blob/c524df017f4afe75633b2ad9d29d087803629e71/src/utils/getRowsInternal.ts#L27)